### PR TITLE
fix: update RPM spec using rpmbuild

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: update RPM spec using rpmbuild

## Motivation and Context
To build RPMs in the future

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#c9a45b4d-812b-41aa-a4e3-250a4adf1d1a) |
<a name="c9a45b4d-812b-41aa-a4e3-250a4adf1d1a"></a>
### 10979-e17f68d/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31720
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 247ed9aa5e60:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 13 seconds
(4/15) Running aws-sdk-ruby tests ... done in 19 seconds
(5/15) Running awscli tests ... done in 3 minutes and 0 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 44 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 41 seconds
(9/15) Running minio-go tests ... done in 6 minutes and 46 seconds
(10/15) Running minio-java tests ... done in 10 minutes and 20 seconds
(11/15) Running minio-js tests ... done in 2 minutes and 39 seconds
(12/15) Running minio-py tests ... done in 18 minutes and 39 seconds
(13/15) Running s3cmd tests ... done in 2 minutes and 13 seconds
(14/15) Running s3select tests ... FAILED in 24 seconds
{
  "name": "s3select:test_csv_output_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-49aa1a9d-8624-4746-8cbf-d46e427b2b6e"
  },
  "duration": 8001,
  "message": "Test test_9 unexpectedly failed with: 'event-type'",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/s3select/csv.py\", line 39, in test_sql_api\n    for d in data.stream(10*1024):\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select/reader.py\", line 211, in stream\n    message = self.__extract_message()\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select/reader.py\", line 161, in __extract_message\n    event_type = header_map[\"event-type\"]\nKeyError: 'event-type'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 56, in main\n    test_csv_output_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 180, in test_csv_output_custom_quote_char\n    input_data, sql_opts, expected_output)\n  File \"/mint/run/core/s3select/csv.py\", line 45, in test_sql_api\n    'Test {} unexpectedly failed with: {}'.format(test_name, select_err))\nValueError: Test test_9 unexpectedly failed with: 'event-type'\n",
  "status": "FAIL"
}
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: minio.spec [hunk 1]
```diff
@@ -1,83 +1,64 @@
...
 rm -rf $RPM_BUILD_ROOT
 
 %files
 %defattr(644,root,root,755)
-%doc README.md README_ZH.md
-%attr(755,root,root) %{_sbindir}/minio
+%attr(644,root,root) /etc/default/minio
+%attr(644,root,root) /etc/systemd/system/minio.service
+%attr(644,minio-user,minio-user) /etc/minio
+%attr(755,minio-user,minio-user) /usr/local/bin/minio
```
Comment (kerneltime) [inline minio.spec:64]: Do you need to prefix the path here with ` $RPM_BUILD_ROOT`

File: minio.spec [hunk 2]
```diff
@@ -1,83 +1,64 @@
...
-v=$(awk '/Commit-ID:/{print $2}' v)
-test "$v" = $commitid
+MINIO_OPTS="--certs-dir /etc/minio/certs"
+EOF
 
-%install
-rm -rf $RPM_BUILD_ROOT
-install -d $RPM_BUILD_ROOT%{_sbindir}
-install -p %{name} $RPM_BUILD_ROOT%{_sbindir}
+install %{_sourcedir}/minio.service $RPM_BUILD_ROOT/etc/systemd/system/minio.service
```
Comment (kerneltime) [inline minio.spec:53]: I am not sure why you need to prefix here with `$RPM_BUILD_ROOT` and if you do need to, then the `attr` setting below would need the same path?

Comment (harshavardhana) [reply minio.spec:53]: this is because whatever is inside RPM_BUILD_ROOT gets packaged as RPM @kerneltime - its `%build` stage

**Review by kannappanr — APPROVED** [review]
LGTM

**Comment by harshavardhana** [comment]
Merging this for Intel IT

**Pull request merged**